src/uu/link/src/link.rs

```diff
@@ -25,16 +53,24 @@ pub fn normalize_error_message(e: Error) -> String {
 }
 
 pub fn uumain(args: impl uucore::Args) -> i32 {
-    let matches = app!(SYNTAX, SUMMARY, LONG_HELP).parse(
-        args.collect_str(InvalidEncodingHandling::Ignore)
-            .accept_any(),
-    );
-    if matches.free.len() != 2 {
-        crash!(1, "{}", msg_wrong_number_of_arguments!(2));
-    }
+    let usage = get_usage();
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about(ABOUT)
+        .usage(&usage[..])
+        .arg(
+            Arg::with_name(options::FILE1)
+                .hidden(true)
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name(options::FILE2)
+                .hidden(true)
+                .takes_value(true),
+        )
```

Comment:
You can set `required(true)` then clap will handle the checking and you don't need `msg_wrong_number_of_arguments!` anymore.
Alternatively you could use only one `Arg::with_name(options::FILE)` with `.multiple(true)` and  `.max_values(2)` and `.min_values(2)`.
And then get the paths like this:
```rust
let files: Vec<String> = matches
    .values_of(option::FILE)
    .map(|v| v.map(ToString::to_string).collect())
    .unwrap_or_default();
    let old = Path::new(&files[0]);
    let new = Path::new(&files[1]);
```

Reply:
Yes, I will change to this format.